Switch to 'console.log' for our hang warning. Add warning to synchronous StackReference calls. (#3456)
pulumi/pulumi: CHANGELOG.md

  calculated by the provider.
  [#3327](https://github.com/pulumi/pulumi/pull/3327)

> - Codepaths which have a high likelihood of causing a hang will print a message to the console

Reviewer: ```suggestion
- Codepaths which could result in a hang will print a message to the console
```
